### DISCUSSION TITLE
Database Introspection / Type Definitions (Serverless Driver)

### DISCUSSION THREAD

**Discussion by stewarthsoj** [planetscale/discussion#281] [Feature Requests] [answered]
I've been playing around with the serverless driver and it's **amazing**! 🎉

I was wondering if database introspection and the ability to generate a TypeScript client are on the roadmap. It'd be awesome if we didn't have to always write the raw SQL queries ourselves. In an ideal world, I'd be able to introspect my PlanetScale database, generate a client and a bunch of type definitions, which gives me a nice TypeScript API to work with my database. This is obviously what the likes of Prisma and Supabase are already doing.

**Comment by jgreet** [comment]
Thanks, glad to hear you're enjoying [database-js](https://github.com/planetscale/database-js). 

My understanding is that at this point we're more interested in [integrating with existing ORMs](https://github.com/planetscale/database-js/issues/50#issuecomment-1224461691) than moving towards developing our own.

**Comment by jgreet** [comment, marked as answer]
This might be of interest https://github.com/depot/kysely-planetscale

**Comment by stewarthsoj** [comment]
Thanks @jgreet! Kysely looks interesting, thanks for the suggestion.
